### PR TITLE
Allow list of connection strings as parameter to AzureAppConfigurationOptions.Connect for Geo-Replication

### PR DESCRIPTION
**Pull request by amerjusupovic**

### CODE REVIEW

**Comment by amerjusupovic** [comment]
Need to wait for pipeline in #388

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs [hunk 1]
```diff
@@ -24,16 +24,18 @@ internal class ConfigurationClientManager : IConfigurationClientManager
...
+                throw new ArgumentNullException(nameof(connectionStrings));
             }
 
-            var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-            var configurationClientWrapper = new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            _clients = new List<ConfigurationClientWrapper> { configurationClientWrapper };
+            _clients = connectionStrings.Select(connectionString => 
+            {
+                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
+                return new ConfigurationClientWrapper(new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection)), new ConfigurationClient(connectionString, clientOptions));
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs:37]: You can use `endpoint` to create new ConfigurationClientWrapper

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs [hunk 2]
```diff
@@ -39,9 +39,9 @@ public class AzureAppConfigurationOptions
         private SortedSet<string> _keyPrefixes = new SortedSet<string>(Comparer<string>.Create((k1, k2) => -string.Compare(k1, k2, StringComparison.OrdinalIgnoreCase)));
 
         /// <summary>
-        /// The connection string to use to connect to Azure App Configuration.
+        /// The list of connection strings used to connect to Azure App Configuration.
```
Comment (zhenlan) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs:42]: Can we reword it so it's clear the connection strings must be for the same store?

**Comment by zhenlan** [comment]
Do we like to change this as well?

https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationExtensions.cs#L22

**Comment by amerjusupovic** [comment]
Should I just add an overload with a list of connection strings?

**Review by zhenlan — APPROVED** [review]

**Review by avanigupta — APPROVED** [review]

**Pull request merged**